Service/EnrichPubliccodeService.php

```diff
@@ -0,0 +1,208 @@
+     */
+    public function enrichRepositoryWithPubliccode(ObjectEntity $repository, array $publiccode = []): ?ObjectEntity
+    {
+        if (!$component = $repository->getValue('component')) {
+            $component = new ObjectEntity($this->componentEntity);
+        }
+
+        if ($publiccode !== null) {
+            if (!$description = $component->getValue('description')) {
+                $description = new ObjectEntity($this->descriptionEntity);
+            }
+
+            $description->hydrate([
+                'shortDescription' => $publiccode['description']['en']['shortDescription'] ?? $publiccode['description']['nl']['shortDescription'] ?? null,
+                'documentation' => $publiccode['description']['en']['documentation'] ?? $publiccode['description']['nl']['documentation'] ?? null,
+                'apiDocumentation' => $publiccode['description']['en']['apiDocumentation'] ?? $publiccode['description']['nl']['apiDocumentation'] ?? null,
+            ]);
+            $this->entityManager->persist($description);
+
+            $component->hydrate([
+                'softwareVersion' => $publiccode['publiccodeYmlVersion'] ?? null,
+                'name' => $publiccode['name'] ?? null,
+                'softwareType' => $publiccode['softwareType'] ?? null,
+                'inputTypes' => $publiccode['inputTypes'] ?? null,
+                'outputTypes' => $publiccode['outputTypes'] ?? null,
+                'platforms' => $publiccode['platforms'] ?? null,
+                'categories' => $publiccode['categories'] ?? null,
+                'developmentStatus' => $publiccode['developmentStatus'] ?? null,
+                'url' => $repository->getId()->toString() ?? null,
+                'description' => $component->getId()->toString(),
+                // Disabled because ? 
+                // 'releaseDate' => $publiccode['releaseDate'] ?? null,
+                // 'dependsOn' => $publiccode['dependsOn']['open']['name'] ?? null,
+                // 'dependsOn' => $publiccode['dependsOn']['open']['versionMin'] ?? null,
+            ]);
+
+            $this->entityManager->persist($component);
+            $repository->setValue('component', $component);
+            $this->entityManager->persist($repository);
+        } else {
+            $component->setValue('name', $repository->getValue('name'));
+            $this->entityManager->persist($component);
+        }
+        $this->entityManager->flush();
+
+        return $repository;
+    }
+
+    /**
+     * This function is searching for repositories containing a publiccode.yaml file. @TODO
+     *
+     * @param string $url
+     *
+     * @throws GuzzleException
+     *
+     * @return array|null|Response
+     */
+    public function getPubliccode(string $url)
+    {
+        $parseUrl = parse_url($url);
+        $code = explode('/blob/', $parseUrl['path']);
+
+        // Find out what the base url is and check sources or create new source?
+        dump($code);die;
```

Comment:
Please do not dump and die

Reply:
Done on purpose because it cant be tested right now. Its fair to say we dont want to merge this right now

Reply:
Maybe add a todo comment and use return instead of die in this case?

Reply:
Also is there an other way to do this besides dump? maybe logging or SymfonyStyle (io write) ?

Reply:
Yup, agreed with the above. No die's schould make it into an pr period.  Als if we cant actually get an publiccode file in te publiccode service then that should defintily be escalated.